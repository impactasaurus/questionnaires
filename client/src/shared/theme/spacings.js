const toREM = value => `${value}rem`;

export default {
    xxs: toREM(.125),
    xs: toREM(.375),
    sm: toREM(.75),
    md: toREM(1.125),
    base: toREM(1.5),
    lg: toREM(1.875),
    xl: toREM(2.25),
    xxl: toREM(2.625),
};