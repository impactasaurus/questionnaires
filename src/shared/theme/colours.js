import { css } from 'styled-components';
const white = 255;
const black = 0;

const toRGB = color => `rgb(${color}, ${color}, ${color})`;
const lerp = (min, max, percent) => min + ((max - min) * percent);

const colours = {
    white: toRGB(white),
    gray20: toRGB(lerp(black, white, .2)),
    gray40: toRGB(lerp(black, white, .4)),
    gray60: toRGB(lerp(black, white, .6)),
    gray80: toRGB(lerp(black, white, .8)),
    green50: 'rgb(157, 198, 107)',
    blue50: 'rgb(67, 97, 194)',
    blue30: 'rgb(79, 164, 154)',
    withAlpha: alpha => value => value.replace(')', `, ${alpha})`).replace('rgb', 'rgba'),
};

export default {
    ...colours,
    primaryGradient: `linear-gradient(45deg, ${colours.green50} 5%, ${colours.blue30} 30%, ${colours.blue50})`,
    defaultStyle: css`
        color: ${props => props.theme.fg};
    `,
};