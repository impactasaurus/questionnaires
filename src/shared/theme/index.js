import rawTypeset from './typeset';
import rawColours from './colours';
import rawSpacings from './spacings';

export const typeset = rawTypeset
export const colours = rawColours;
export const spacings = rawSpacings;

export const primaryTheme = {
    fg: colours.gray60,
    bg: 'transparent',
    highlight: colours.green50,
    lowlight: colours.white,
    gradient: colours.primaryGradient,
};

export const secondaryTheme = {
    fg: colours.white,
    bg: colours.blue30,
    highlight: colours.white,
    lowlight: 'transparent',
    gradient: colours.primaryGradient,
};