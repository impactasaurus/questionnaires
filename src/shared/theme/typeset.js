import { css } from 'styled-components';

const toREM = value => `${value}rem`;
const scale = by => value => value * by;
const lineHeightScale = scale(1.5);

const sizes = {
    xs: .5,
    sm: .75,
    base: 1,
    lg: 1.5,
    xl: 2.25,
    xxl: 2.625,
}

const mapSizes = (mappingFn, initial) => Object.keys(sizes)
    .reduce((acc, size) => ({
        ...acc,
        [size]: mappingFn(sizes[size]),
    }), initial);

const typeset = {
    fontFamily: `"Source Sans Pro", Helvetica, sans-serif`,
    size: mapSizes(toREM, { root: '100%' }),
    lineHeight: mapSizes(s => toREM(lineHeightScale(s)), {}),
    weight: {
        hairline: 100,
        extraLight: 200,
        light: 300,
        base: 400,
        medium: 500,
        semiBold: 600,
        bold: 700,
        extraBold: 800,
        heavy: 900,
    }
};

export default {
    ...typeset,
    defaultStyle: css`
        font-family: ${typeset.fontFamily};
        font-size: ${typeset.size.base};
        line-height: ${typeset.lineHeight.base};
        font-weight: ${typeset.weight.light};
    `,
};