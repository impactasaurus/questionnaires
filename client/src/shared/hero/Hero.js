import React, {Component} from 'react';
import styled, { ThemeProvider } from 'styled-components';
import {  secondaryTheme } from '../theme';
import Header, { Subheader } from '../header';
import firstLayerOverlay from './overlay.png';
import secondLayerOverlay from './overlay.svg';

export const Section = styled.section`
    background-color: ${props => props.theme.bg};
    background-image: url(${firstLayerOverlay}), url(${secondLayerOverlay}), ${props => props.theme.gradient};
    background-position: top left, center center, center center;
    background-attachment: fixed, fixed, fixed;
    background-size: auto, cover, cover;
    background-repeat: no-repeat;
    display: flex;
    flex-flow: row nowrap;
    justify-content: center;
    min-height: 24em;
`;

export const Inner = styled.div`
    display: flex;
    flex-flow: column nowrap; 
    justify-content: ${props => props.hasActions ? 'flex-end' : 'center'}};
    align-items: center;
    transform: scale(${props => props.visible ? 1 : 1.1});
    opacity: ${props => props.visible ? 1 : 0};
    transition-delay: 0.25s;
    transition: transform 1.5s ease, opacity 2s ease;
`;

export default class Hero extends Component {
    constructor(props) {
        super(props);
        this.state = {
            visible: false,
        };
    }

    componentDidMount() {
        // We require a timeout as this method is called
        // before react has actually rendered the VirtualDOM 
        // to the real DOM. As a result, if we don't delay
        // this then the first render of the real DOM will 
        // have state.visible = true, and so we won't get our 
        // CSS transitions.
        window.setTimeout(() => {
            this.setState({
                visible: true,
            });
        }, 0);
    }

    render() {
        return (
            <ThemeProvider theme={secondaryTheme}>
                <Section>
                    <Inner {...this.state} hasActions={!!React.Children.count(this.props.children)}>
                        <Header>Impactasaurus Questionnaire Repository</Header>
                        <Subheader>Find a questionnaire to demonstrate and improve your social impact</Subheader>
                        {this.props.children}
                    </Inner>
                </Section>
            </ThemeProvider>
        );
    }
}