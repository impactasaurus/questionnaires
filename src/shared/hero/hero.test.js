import React from 'react';
import renderer from 'react-test-renderer';
import { ThemeProvider } from 'styled-components';
import { secondaryTheme } from '../theme';
import Hero, { Section, Inner } from './Hero';

describe('<Hero />', () => {
    const theme = {
        bg: '#bbb000',
        gradient: 'linear-gradient(45deg,#bbb000 0%,#fff000)'
    };

    const render = component => renderer.create(
        <ThemeProvider theme={theme}>
            {component}
        </ThemeProvider>
    );

    const renderToJson = component => render(component).toJSON();

    describe('<Section />', () => {
        it('should respect the theme background colour', () => {
            const component = renderToJson(<Section />);
            expect(component).toHaveStyleRule('background-color', theme.bg);
            expect(component).toHaveStyleRule('background-image', new RegExp(theme.gradient.replace('(', '\\(').replace(')', '\\)')));
        });
    });

    describe('<Inner />', () => {
        it('should be centered if there are no actions', () => {
            const component = renderToJson(<Inner />);
            expect(component).toHaveStyleRule('justify-content', 'center');
        });

        it('should be justfied to the end if there are actions', () => {
            const component = renderToJson(<Inner hasActions />);
            expect(component).toHaveStyleRule('justify-content', 'flex-end');
        });

        it('should have no opacity and slightly larger scale when not visible', () => {
            const component = renderToJson(<Inner />);
            expect(component).toHaveStyleRule('opacity', '0');
            expect(component).toHaveStyleRule('transform', 'scale(1.1)');
        });

        it('should have full opacity and normal scale when visible', () => {
            const component = renderToJson(<Inner visible />);
            expect(component).toHaveStyleRule('opacity', '1');
            expect(component).toHaveStyleRule('transform', 'scale(1)');
        });
    });

    it('should always use the secondary theme', () => {
        const component = renderer.create(<Hero />);
        expect(component.root.findByType(ThemeProvider).props.theme).toEqual(secondaryTheme);
    });

    it('should render an invisible inner section', () => {
        const component = renderer.create(<Hero />);
        expect(component.root.findByType(Inner).props.visible).toBeFalsy();
    });

    it('should mark the inner section as having no actions when no children are given to the hero', () => {
        const component = renderer.create(<Hero />);
        expect(component.root.findByType(Inner).props.hasActions).toBeFalsy();
    });

    it('should mark the inner section as having actions when children are given to the hero', () => {
        const component = renderer.create(<Hero><p>Action!</p></Hero>);
        expect(component.root.findByType(Inner).props.hasActions).toBeTruthy();
    });

    it('should mark the inner section as visible immediately after mounting', () => {
        jest.useFakeTimers();
        const component = renderer.create(<Hero />);
        jest.runTimersToTime(1);
        expect(component.root.findByType(Inner).props.visible).toBeTruthy();
    });

    it('should match the snapshot', () => {
        const component = renderer.create(<Hero><p>Action!</p></Hero>);
        expect(component).toMatchSnapshot();
    });
});