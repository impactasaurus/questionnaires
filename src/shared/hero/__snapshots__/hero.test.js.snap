// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Hero /> should match the snapshot 1`] = `
.c2 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  color: rgb(255,255,255);
  line-height: 3.375rem;
  font-size: 2.25rem;
  font-weight: 300;
  margin: 0;
}

.c3 {
  font-family: "Source Sans Pro",Helvetica,sans-serif;
  color: rgb(255,255,255);
  line-height: 1.5rem;
  font-size: 1rem;
  font-weight: 300;
  margin: 0;
  margin-top: 1.125rem;
}

.c0 {
  background-color: rgb(79,164,154);
  background-image: url(overlay.png),url(overlay.svg),linear-gradient(45deg,rgb(157,198,107) 5%,rgb(79,164,154) 30%,rgb(67,97,194));
  background-position: top left,center center,center center;
  background-attachment: fixed,fixed,fixed;
  background-size: auto,cover,cover;
  background-repeat: no-repeat;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: row nowrap;
  -ms-flex-flow: row nowrap;
  flex-flow: row nowrap;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  min-height: 24em;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-transform: scale(1.1);
  -ms-transform: scale(1.1);
  transform: scale(1.1);
  opacity: 0;
  -webkit-transition-delay: 0.25s;
  transition-delay: 0.25s;
  -webkit-transition: -webkit-transform 1.5s ease,opacity 2s ease;
  -webkit-transition: transform 1.5s ease,opacity 2s ease;
  transition: transform 1.5s ease,opacity 2s ease;
}

<section
  className="c0"
>
  <div
    className="c1"
  >
    <h1
      className="c2"
    >
      Impactasaurus Questionnaire Repository
    </h1>
    <p
      className="c3"
    >
      Find a questionnaire to demonstrate and improve your social impact
    </p>
    <p>
      Action!
    </p>
  </div>
</section>
`;
